diff -urN Youtube-Reborn/Controllers/RootOptionsController.m Youtube-Reborn+NativeShare/Controllers/RootOptionsController.m
--- Youtube-Reborn/Controllers/RootOptionsController.m	2022-03-15 15:15:30.560370445 +0700
+++ Youtube-Reborn+NativeShare/Controllers/RootOptionsController.m	2022-03-15 22:03:53.603308337 +0700
@@ -64,7 +64,7 @@
         return 7;
     }
     if (section == 3) {
-        return 8;
+        return 9;
     }
     if (section == 4) {
         return 2;
@@ -202,6 +202,14 @@
                 hideYouTubeLogo.on = [[NSUserDefaults standardUserDefaults] boolForKey:@"kHideYouTubeLogo"];
                 cell.accessoryView = hideYouTubeLogo;
             }
+            if(indexPath.row == 8) {
+                cell.textLabel.text = @"Use Native Share Sheet";
+                cell.selectionStyle = UITableViewCellSelectionStyleNone;
+                UISwitch *useNativeShareSheet = [[UISwitch alloc] initWithFrame:CGRectZero];
+                [useNativeShareSheet addTarget:self action:@selector(toggleUseNativeShareSheet:) forControlEvents:UIControlEventValueChanged];
+                useNativeShareSheet.on = [[NSUserDefaults standardUserDefaults] boolForKey:@"kUseNativeShareSheet"];
+                cell.accessoryView = useNativeShareSheet;
+            }
         }
         if(indexPath.section == 4) {
             if(indexPath.row == 0) {
@@ -449,5 +457,15 @@
         [[NSUserDefaults standardUserDefaults] synchronize];
     }
 }
+
+- (void)toggleUseNativeShareSheet:(UISwitch *)sender {
+    if ([sender isOn]) {
+        [[NSUserDefaults standardUserDefaults] setBool:YES forKey:@"kUseNativeShareSheet"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+    } else {
+        [[NSUserDefaults standardUserDefaults] setBool:NO forKey:@"kUseNativeShareSheet"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+    }
+}
 
 @end
diff -urN Youtube-Reborn/Tweak.h Youtube-Reborn+NativeShare/Tweak.h
--- Youtube-Reborn/Tweak.h	2022-03-15 15:15:30.596944387 +0700
+++ Youtube-Reborn+NativeShare/Tweak.h	2022-03-15 21:56:47.941196142 +0700
@@ -77,6 +77,17 @@
 - (void)rootOptionsAction:(id)sender;
 @end
 
+@interface YTIShareTargetServiceUpdateRenderer
+- (int)serviceId;
+@end
+
+@interface YTShareTargetCell: UIView
+- (void)didTap;
+@end
+
+@interface YTShareRequestViewController: UIViewController
+@end
+
 // Tweak Variables
 
 extern NSMutableArray *overlayButtons;
diff -urN Youtube-Reborn/Tweak.xm Youtube-Reborn+NativeShare/Tweak.xm
--- Youtube-Reborn/Tweak.xm	2022-03-15 15:15:30.599266498 +0700
+++ Youtube-Reborn+NativeShare/Tweak.xm	2022-03-15 22:00:17.076075743 +0700
@@ -1683,6 +1683,37 @@
 %end
 %end
 
+%group gUseNativeShareSheet
+%hook YTShareTargetCell
+
+- (void)drawRect:(CGRect)rect {
+	%orig;
+
+	UIView* superview = self.superview;
+
+	if ([superview isKindOfClass: [UICollectionView class]]) {
+		UICollectionView* collectionView = (UICollectionView*)superview;
+	    [collectionView setContentOffset: CGPointMake([collectionView contentSize].width - collectionView.frame.size.width, 0) animated: NO];
+	}
+}
+
+- (void)setEntry:(YTIShareTargetServiceUpdateRenderer*)entry {
+	%orig;
+	if ([entry serviceId] != 82) {
+		return;
+	}
+	[self didTap];
+}
+%end
+
+%hook YTShareRequestViewController
+- (void)viewDidLoad {
+	[self.view setAlpha: 0.0];
+}
+%end
+%end
+
+
 int selectedTabIndex = 0;
 
 %hook YTPivotBarViewController
@@ -1821,6 +1852,7 @@
         if ([[NSUserDefaults standardUserDefaults] boolForKey:@"kHideYouTubeLogo"] == YES) %init(gHideYouTubeLogo);
         if ([[NSUserDefaults standardUserDefaults] boolForKey:@"kEnableEnhancedSearchBar"] == YES) %init(gEnableEnhancedSearchBar);
         if ([[NSUserDefaults standardUserDefaults] boolForKey:@"kHideTabBar"] == YES) %init(gHideTabBar);
+        if ([[NSUserDefaults standardUserDefaults] boolForKey:@"kUseNativeShareSheet"] == YES) %init(gUseNativeShareSheet);
         if ([[NSUserDefaults standardUserDefaults] boolForKey:@"kEnableiPadStyleOniPhone"] == NO & [[NSUserDefaults standardUserDefaults] boolForKey:@"kShowStatusBarInOverlay"] == YES) {
             %init(gShowStatusBarInOverlay);
         }
