diff -urN Youtube-Reborn/Controllers/DownloadsController.h Youtube-Reborn+DoneButtonInDownloadController/Controllers/DownloadsController.h
--- Youtube-Reborn/Controllers/DownloadsController.h	2022-03-02 09:27:53.000000000 +0700
+++ Youtube-Reborn+DoneButtonInDownloadController/Controllers/DownloadsController.h	2022-03-16 10:35:45.530590424 +0700
@@ -4,4 +4,4 @@
 
 @property (nonatomic, retain) UITabBarController *tabBar;
 
-@end
\ No newline at end of file
+@end
diff -urN Youtube-Reborn/Controllers/DownloadsController.m Youtube-Reborn+DoneButtonInDownloadController/Controllers/DownloadsController.m
--- Youtube-Reborn/Controllers/DownloadsController.m	2022-03-02 09:27:53.000000000 +0700
+++ Youtube-Reborn+DoneButtonInDownloadController/Controllers/DownloadsController.m	2022-03-16 17:48:22.027855605 +0700
@@ -9,21 +9,39 @@
 @end
 
 @implementation DownloadsController
-
 - (void)loadView {
 	[super loadView];
 
     [self.navigationController setNavigationBarHidden:YES animated:NO];
-
     self.tabBar = [[UITabBarController alloc] init];
 
     DownloadsVideoController *videoViewController = [[DownloadsVideoController alloc] init];
     videoViewController.title = @"Video";
+    videoViewController.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(done)];;
+    UINavigationController *videoViewNavigation = [[UINavigationController alloc] initWithRootViewController:videoViewController];
 
     DownloadsAudioController *audioViewController = [[DownloadsAudioController alloc] init];
     audioViewController.title = @"Audio";
+    audioViewController.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(done)];
+    UINavigationController *audioViewNavigation = [[UINavigationController alloc] initWithRootViewController:audioViewController];
+
+    if (self.traitCollection.userInterfaceStyle == UIUserInterfaceStyleLight) {
+        self.view.backgroundColor = [UIColor colorWithRed:0.949 green:0.949 blue:0.969 alpha:1.0];
+        [videoViewNavigation.navigationBar setTitleTextAttributes:@{NSForegroundColorAttributeName:[UIColor blackColor]}];
+        [audioViewNavigation.navigationBar setTitleTextAttributes:@{NSForegroundColorAttributeName:[UIColor blackColor]}];
+    }
+    else {
+        self.view.backgroundColor = [UIColor colorWithRed:0.0 green:0.0 blue:0.0 alpha:1.0];
+        [videoViewNavigation.navigationBar setTitleTextAttributes:@{NSForegroundColorAttributeName:[UIColor whiteColor]}];
+        [audioViewNavigation.navigationBar setTitleTextAttributes:@{NSForegroundColorAttributeName:[UIColor whiteColor]}];
+        videoViewNavigation.navigationBar.barStyle = UIBarStyleBlack;
+        audioViewNavigation.navigationBar.barStyle = UIBarStyleBlack;
+    }
 
-    self.tabBar.viewControllers = [NSArray arrayWithObjects:videoViewController, audioViewController, nil];
+    videoViewNavigation.tabBarItem = [[UITabBarItem alloc] initWithTitle:@"Video" image:[UIImage systemImageNamed:@"film"] selectedImage:[UIImage systemImageNamed:@"film.fill"]];
+    audioViewNavigation.tabBarItem = [[UITabBarItem alloc] initWithTitle:@"Audio" image:[UIImage systemImageNamed:@"music.note"] tag:0];
+
+    self.tabBar.viewControllers = [NSArray arrayWithObjects:videoViewNavigation, audioViewNavigation, nil];
 
     [self.view addSubview:self.tabBar.view];
 }
@@ -43,4 +60,4 @@
     [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
 }
 
-@end
\ No newline at end of file
+@end
