diff -urN Youtube-Reborn/Controllers/RootOptionsController.m Youtube-Reborn+showDownloadsInFiles/Controllers/RootOptionsController.m
--- Youtube-Reborn/Controllers/RootOptionsController.m	2022-03-16 19:25:09.457151950 +0700
+++ Youtube-Reborn+showDownloadsInFiles/Controllers/RootOptionsController.m	2022-03-16 19:50:48.049166159 +0700
@@ -54,11 +54,9 @@
         return 1;
     }
     if (section == 1) {
-        if ([DTTJailbreakDetection isJailbroken]) {
-            return 2;
-        } else {
-            return 1;
-        }
+        BOOL canOpenInFiles = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UISupportsDocumentBrowser"];
+        BOOL canOpenInFilza = [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:@"filza://"]];
+        return 1 + canOpenInFiles + canOpenInFilza;
     }
     if (section == 2) {
         return 7;
@@ -98,9 +96,32 @@
                 cell.textLabel.text = @"View Downloads";
                 cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
             }
-            if(indexPath.row == 1) {
-                cell.textLabel.text = @"View Downloads In Filza";
-                cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+            if ([[NSBundle mainBundle] objectForInfoDictionaryKey:@"UISupportsDocumentBrowser"] && [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:@"filza://"]]) {
+                // Can open in both Files and Filza (probably)
+                if (indexPath.row == 1) {
+                    cell.textLabel.text = @"View Downloads In Files";
+                    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+                }
+                if (indexPath.row == 2) {
+                    cell.textLabel.text = @"View Downloads In Filza";
+                    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+                }
+            else if ([[NSBundle mainBundle] objectForInfoDictionaryKey:@"UISupportsDocumentBrowser"]) {
+                // Can only open in Files app
+                if (indexPath.row == 1) {
+                    cell.textLabel.text = @"View Downloads In Files";
+                    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+                }
+            }
+            else if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:@"filza://"]]) {
+                // Can only open in Filza
+                if (indexPath.row == 1) {
+                    cell.textLabel.text = @"View Downloads In Filza";
+                    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
+                }
+            }
+            else {
+                // If you reach this place, then I guess View Downloads is your only option
             }
         }
         if(indexPath.section == 2) {
@@ -240,16 +261,38 @@
 
             [self presentViewController:downloadsControllerView animated:YES completion:nil];
         }
-        if(indexPath.row == 1) {
-            NSFileManager * fm = [[NSFileManager alloc] init];
+        if (indexPath.row == 1 || indexPath.row == 2) {
+            NSFileManager *fm = [[NSFileManager alloc] init];
             NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
             NSString *documentsDirectory = [paths objectAtIndex:0];
-            NSString *content = @"Filza Check";
-            NSData *fileContents = [content dataUsingEncoding:NSUTF8StringEncoding];
-            NSString *filePath = [documentsDirectory stringByAppendingPathComponent:@"FilzaCheck.txt"];
-            [fm createFileAtPath:filePath contents:fileContents attributes:nil];
-            NSString *path = [NSString stringWithFormat:@"filza://view%@/FilzaCheck.txt", documentsDirectory];
-            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:path] options:@{} completionHandler:nil];
+            BOOL canOpenInFiles = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UISupportsDocumentBrowser"];
+            BOOL canOpenInFilza = [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:@"filza://"]];
+            
+            if (canOpenInFiles && canOpenInFilza) {
+                if (indexPath.row == 1) {
+                    NSString *path = [NSString stringWithFormat:@"shareddocuments://%@", documentsDirectory];
+                    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:path] options:@{} completionHandler:nil];
+                }
+                if (indexPath.row == 2) {
+                    NSString *path = [NSString stringWithFormat:@"filza://view%@", documentsDirectory];
+                    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:path] options:@{} completionHandler:nil];
+                }
+            }
+            else if (canOpenInFiles) {
+                if (indexPath.row == 1) {
+                    NSString *path = [NSString stringWithFormat:@"shareddocuments://%@", documentsDirectory];
+                    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:path] options:@{} completionHandler:nil];
+                }
+            }
+            else if (canOpenInFilza) {
+                if (indexPath.row == 1) {
+                    NSString *path = [NSString stringWithFormat:@"filza://view%@", documentsDirectory];
+                    [[UIApplication sharedApplication] openURL:[NSURL URLWithString:path] options:@{} completionHandler:nil];
+                }
+            }
+            else {
+                // If you reach this place, then I guess View Downloads is your only option
+            }
         }
     }
     if(indexPath.section == 2) {
