diff -urN Youtube-Reborn/Tweak.xm Youtube-Reborn+APIKey/Tweak.xm
--- Youtube-Reborn/Tweak.xm	2022-03-02 09:27:53.000000000 +0700
+++ Youtube-Reborn+APIKey/Tweak.xm	2022-03-10 20:35:31.014110325 +0700
@@ -8,7 +8,7 @@
 #import "DTTJailbreakDetection/DTTJailbreakDetection.h"
 #import "Tweak.h"
 
-NSString *YTApiKey = @"AIzaSyAcHI73xntvCiURGWERFcJxm5vX12p5bN8";
+NSString *YTApiKey = @"AIzaSyCGZHGUPP9wojQDfrwPEr7Pr-mhBanHAb8";
 
 UIColor *hexColour() {
     NSData *colorData = [[NSUserDefaults standardUserDefaults] objectForKey:@"kYTRebornColourOptionsVTwo"];
@@ -1842,4 +1842,4 @@
         }
         %init(_ungrouped);
     }
-}
\ No newline at end of file
+}
diff -urN Youtube-Reborn/XCDYouTubeKit/XCDYouTubeClient.m Youtube-Reborn+APIKey/XCDYouTubeKit/XCDYouTubeClient.m
--- Youtube-Reborn/XCDYouTubeKit/XCDYouTubeClient.m	2022-03-02 09:27:53.000000000 +0700
+++ Youtube-Reborn+APIKey/XCDYouTubeKit/XCDYouTubeClient.m	2022-03-10 20:36:05.071923402 +0700
@@ -14,7 +14,7 @@
 
 @synthesize languageIdentifier = _languageIdentifier;
 
-static NSString * _innertubeApiKey = @"AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8";
+static NSString * _innertubeApiKey = @"AIzaSyCGZHGUPP9wojQDfrwPEr7Pr-mhBanHAb8";
 
 + (instancetype) defaultClient
 {
