--- YTETA/Tweak.xm	2022-03-15 21:06:48.907210083 +0700
+++ YTETA+patch/Tweak.xm	2022-03-15 21:13:16.166162302 +0700
@@ -1,3 +1,4 @@
+#import <Foundation/Foundation.h>
 #import <UIKit/UIKit.h>
 
 @interface YTPlayerViewController : UIViewController
@@ -17,13 +18,6 @@
 static BOOL secondsEnabled;
 static BOOL twentyFourHourClockEnabled;
 
-static void loadPrefs() {
-	NSMutableDictionary *prefs = [[NSMutableDictionary alloc] initWithContentsOfFile:@"/var/mobile/Library/Preferences/com.icraze.ytetaprefs.plist"];
-	enabled = [prefs objectForKey:@"enabled"] ? [[prefs objectForKey:@"enabled"] boolValue] : YES;
-	secondsEnabled = [prefs objectForKey:@"secondsEnabled"] ? [[prefs objectForKey:@"secondsEnabled"] boolValue] : YES;
-	twentyFourHourClockEnabled = [prefs objectForKey:@"twentyFourHourClockEnabled"] ? [[prefs objectForKey:@"twentyFourHourClockEnabled"] boolValue] : NO;
-}
-
 static void modifyLabel(UILabel *label, float remainingSeconds, float videoSpeed) {
 	NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
 	NSMutableString *formatString = [[NSMutableString alloc] initWithString:@"mm"];
@@ -60,7 +54,8 @@
 %end
 
 %ctor {
-	loadPrefs();
-	CFNotificationCenterAddObserver(CFNotificationCenterGetDarwinNotifyCenter(), NULL, (CFNotificationCallback)loadPrefs, CFSTR("com.icraze.ytetaprefs.settingschanged"), NULL, CFNotificationSuspensionBehaviorCoalesce);
+	enabled = [[NSUserDefaults standardUserDefaults] boolForKey:@"kShowWhenVideoEnds"];
+	secondsEnabled = [[NSUserDefaults standardUserDefaults] boolForKey:@"kShowSeconds"];
+	twentyFourHourClockEnabled = [[NSUserDefaults standardUserDefaults] boolForKey:@"kUse24HourClock"];
 	if (enabled) %init;
 }
