--- YTETA/Makefile	2022-03-15 21:06:48.907023727 +0700
+++ YTETA+patch/Makefile	2022-03-15 21:13:23.213625692 +0700
@@ -1,6 +1,5 @@
-export PREFIX = $(THEOS)/toolchain/Xcode11.xctoolchain/usr/bin/
-export SDKVERSION = 14.4
-export ARCHS = arm64 arm64e
+TARGET := iphone:clang:14.5:13.0
+ARCHS = arm64 arm64e
 
 INSTALL_TARGET_PROCESSES = YouTube
 
@@ -10,7 +9,6 @@
 
 YTETA_FILES = Tweak.xm
 YTETA_CFLAGS = -fobjc-arc
+YTETA_FRAMEWORKS = Foundation UIKit
 
 include $(THEOS_MAKE_PATH)/tweak.mk
-SUBPROJECTS += ytetaprefs
-include $(THEOS_MAKE_PATH)/aggregate.mk
