--- YTSpeed/Makefile	2022-03-15 14:14:57.507860264 +0700
+++ YTSpeed+makefile/Makefile	2022-03-15 14:42:33.251616430 +0700
@@ -1,6 +1,7 @@
-TARGET := iphone:clang:latest:7.0
+TARGET := iphone:clang:14.5:13:0
 INSTALL_TARGET_PROCESSES = YouTube
-
+GO_EASY_ON_ME = 1
+ARCHS = arm64 arm64e
 
 include $(THEOS)/makefiles/common.mk
 
@@ -8,5 +9,6 @@
 
 YoutubeSpeed_FILES = Tweak.xm
 YoutubeSpeed_CFLAGS = -fobjc-arc
+YoutubeSpeed_FRAMEWORKS = Foundation
 
 include $(THEOS_MAKE_PATH)/tweak.mk
