diff -urN iSponsorBlock/SponsorBlockSettingsController.m iSponsorBlock+jailed/SponsorBlockSettingsController.m
--- iSponsorBlock/SponsorBlockSettingsController.m	2022-03-10 16:51:13.041403708 +0700
+++ iSponsorBlock+jailed/SponsorBlockSettingsController.m	2022-03-10 16:51:01.958502337 +0700
@@ -1,6 +1,8 @@
 #import "SponsorBlockSettingsController.h"
 #import "colorFunctions.h"
 
+static NSString *LogoSponsorBlockerPath;
+
 @implementation SponsorBlockTableCell
 -(void)colorPicker:(id)colorPicker didSelectColor:(UIColor *)color {
     self.colorWell.color = color;
@@ -62,7 +64,16 @@
     self.tableView.delegate = self;
     self.tableView.dataSource = self;
     
-    UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/LogoSponsorBlocker128px.png"]];
+    NSString *resourcesBundlePath = [[NSBundle mainBundle] pathForResource:@"iSponsorBlock" ofType:@"bundle"];
+    if (resourcesBundlePath) {
+       NSBundle *resourcesBundle = [NSBundle bundleWithPath:resourcesBundlePath];
+       LogoSponsorBlockerPath = [resourcesBundle pathForResource:@"LogoSponsorBlocker128px" ofType:@"png"]; 
+    } else {
+       LogoSponsorBlockerPath = @"/Library/Application Support/iSponsorBlock.bundle/LogoSponsorBlocker128px.png";    
+    }
+ 
+    UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageWithContentsOfFile:LogoSponsorBlockerPath]];;
     UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake(0,0,0,0)];
     label.text = @"iSponsorBlock";
     label.font = [UIFont boldSystemFontOfSize:48];
diff -urN iSponsorBlock/iSponsorBlock.xm iSponsorBlock+jailed/iSponsorBlock.xm
--- iSponsorBlock/iSponsorBlock.xm	2022-03-10 16:51:13.043755864 +0700
+++ iSponsorBlock+jailed/iSponsorBlock.xm	2022-03-10 16:51:01.962073447 +0700
@@ -4,6 +4,11 @@
 #import "SponsorBlockRequest.h"
 #import "SponsorBlockViewController.h"
 
+static NSString *PlayerInfoIconSponsorBlockerPath;
+static NSString *SponsorblockendPath;
+static NSString *SponsorblockstartPath;
+static NSString *SponsorblocksettingsPath;
+
 %group Main
 NSString *modifiedTimeString;
 
@@ -214,15 +219,17 @@
     NSArray <UIView *> *topControls = %orig;
     if(![topControls containsObject:self.sponsorBlockButton] && kShowButtonsInPlayer){
         NSMutableArray *mutableArray = topControls.mutableCopy;
         if(!self.sponsorBlockButton){
             self.sponsorBlockButton = [%c(YTQTMButton) iconButton];
             self.sponsorBlockButton.frame = CGRectMake(0, 0, 24, 36);
-            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/PlayerInfoIconSponsorBlocker256px-20@2x.png"] forState:UIControlStateNormal];
+            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:PlayerInfoIconSponsorBlockerPath] forState:UIControlStateNormal];
             
             self.sponsorStartedEndedButton = [%c(YTQTMButton) iconButton];
             self.sponsorStartedEndedButton.frame = CGRectMake(0,0,24,36);
-            if(self.playerViewController.userSkipSegments.lastObject.endTime != -1) [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockstart-20@2x.png"] forState:UIControlStateNormal];
-            else [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockend-20@2x.png"] forState:UIControlStateNormal];
+            if(self.playerViewController.userSkipSegments.lastObject.endTime != -1) [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SponsorblockstartPath] forState:UIControlStateNormal];
+            else [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SponsorblockendPath] forState:UIControlStateNormal];
 
             if(topControls[0].superview == self){
                 [self addSubview:self.sponsorBlockButton];
@@ -278,9 +285,11 @@
 }
 %new
 -(void)sponsorStartedEndedButtonPressed:(YTQTMButton *)sender {
     if(self.playerViewController.userSkipSegments.lastObject.endTime != -1) {
         [self.playerViewController.userSkipSegments addObject:[[SponsorSegment alloc] initWithStartTime:self.playerViewController.currentVideoMediaTime endTime:-1 category:nil UUID:nil]];
-       [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockend-20@2x.png"] forState:UIControlStateNormal];
+        [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SponsorblockendPath] forState:UIControlStateNormal];
     }
     else {
         self.playerViewController.userSkipSegments.lastObject.endTime = self.playerViewController.currentVideoMediaTime;
@@ -292,7 +301,7 @@
             [[[UIApplication sharedApplication] delegate].window.rootViewController  presentViewController:alert animated:YES completion:nil];
             return;
         }
-        [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockstart-20@2x.png"] forState:UIControlStateNormal];
+        [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SponsorblockstartPath] forState:UIControlStateNormal];
     }
 }
 %new
@@ -711,10 +722,10 @@
         self.sponsorBlockButton.frame = CGRectMake(0, 0, 40, 40);
         
         if([%c(YTPageStyleController) pageStyle]) { //dark mode
-            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblocksettings-20@2x.png"] forState:UIControlStateNormal];
+            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:SponsorblocksettingsPath] forState:UIControlStateNormal];
         }
         else { //light mode
-            UIImage *image = [UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblocksettings-20@2x.png"];
+            UIImage *image = [UIImage imageWithContentsOfFile:SponsorblocksettingsPath];
             image = [image imageWithTintColor:UIColor.blackColor renderingMode:UIImageRenderingModeAlwaysTemplate];
             [self.sponsorBlockButton setImage:image forState:UIControlStateNormal];
             [self.sponsorBlockButton setTintColor:UIColor.blackColor];
@@ -839,3 +850,24 @@
         }
     }
 }
+
+
+//Bundle
+
+%ctor {
+NSString *resourcesBundlePath = [[NSBundle mainBundle] pathForResource:@"iSponsorBlock" ofType:@"bundle"];
+    if (resourcesBundlePath) {
+    NSBundle *resourcesBundle = [NSBundle bundleWithPath:resourcesBundlePath];
+    PlayerInfoIconSponsorBlockerPath = [resourcesBundle pathForResource:@"PlayerInfoIconSponsorBlocker256px-20@2x" ofType:@"png"];
+    SponsorblockstartPath = [resourcesBundle pathForResource:@"sponsorblockstart-20@2x" ofType:@"png"];
+    SponsorblockendPath = [resourcesBundle pathForResource:@"sponsorblockend-20@2x" ofType:@"png"];
+    SponsorblocksettingsPath = [resourcesBundle pathForResource:@"sponsorblocksettings-20@2x" ofType:@"png"];
+
+    } else {
+    PlayerInfoIconSponsorBlockerPath = @"/Library/Application Support/iSponsorBlock.bundle/PlayerInfoIconSponsorBlocker256px-20@2x.png";    
+    SponsorblockstartPath = @"/Library/Application Support/iSponsorBlock.bundle/sponsorblockstart-20@2x.png"; 
+    SponsorblockendPath = @"/Library/Application Support/iSponsorBlock.bundle/sponsorblockend-20@2x.png"; 
+    SponsorblocksettingsPath = @"/Library/Application Support/iSponsorBlock.bundle/sponsorblocksettings-20@2x.png"; 
+    }
+    %init;
+}
