diff -urN iSponsorBlock/iSponsorBlock.xm iSponsorBlock+jailed/iSponsorBlock.xm
--- iSponsorBlock/iSponsorBlock.xm	2022-03-10 18:25:44.436723531 +0700
+++ iSponsorBlock+jailed/iSponsorBlock.xm	2022-03-10 18:34:14.061899970 +0700
@@ -4,6 +4,11 @@
 #import "SponsorBlockRequest.h"
 #import "SponsorBlockViewController.h"
 
+static NSString *PlayerInfoIconSponsorBlockerPath;
+static NSString *SBEndPath;
+static NSString *SBStartPath;
+static NSString *SBSettingsPath;
+
 %group Main
 NSString *modifiedTimeString;
 
@@ -217,12 +222,12 @@
         if(!self.sponsorBlockButton){
             self.sponsorBlockButton = [%c(YTQTMButton) iconButton];
             self.sponsorBlockButton.frame = CGRectMake(0, 0, 24, 36);
-            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/PlayerInfoIconSponsorBlocker256px-20@2x.png"] forState:UIControlStateNormal];
+            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:PlayerInfoIconSponsorBlockerPath] forState:UIControlStateNormal];
             
             self.sponsorStartedEndedButton = [%c(YTQTMButton) iconButton];
             self.sponsorStartedEndedButton.frame = CGRectMake(0,0,24,36);
-            if(self.playerViewController.userSkipSegments.lastObject.endTime != -1) [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockstart-20@2x.png"] forState:UIControlStateNormal];
-            else [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockend-20@2x.png"] forState:UIControlStateNormal];
+            if(self.playerViewController.userSkipSegments.lastObject.endTime != -1) [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SBStartPath] forState:UIControlStateNormal];
+            else [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SBEndPath] forState:UIControlStateNormal];
 
             if(topControls[0].superview == self){
                 [self addSubview:self.sponsorBlockButton];
@@ -280,7 +285,7 @@
 -(void)sponsorStartedEndedButtonPressed:(YTQTMButton *)sender {
     if(self.playerViewController.userSkipSegments.lastObject.endTime != -1) {
         [self.playerViewController.userSkipSegments addObject:[[SponsorSegment alloc] initWithStartTime:self.playerViewController.currentVideoMediaTime endTime:-1 category:nil UUID:nil]];
-       [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockend-20@2x.png"] forState:UIControlStateNormal];
+        [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SBEndPath] forState:UIControlStateNormal];
     }
     else {
         self.playerViewController.userSkipSegments.lastObject.endTime = self.playerViewController.currentVideoMediaTime;
@@ -292,7 +297,7 @@
             [[[UIApplication sharedApplication] delegate].window.rootViewController  presentViewController:alert animated:YES completion:nil];
             return;
         }
-        [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblockstart-20@2x.png"] forState:UIControlStateNormal];
+        [self.sponsorStartedEndedButton setImage:[UIImage imageWithContentsOfFile:SBStartPath] forState:UIControlStateNormal];
     }
 }
 %new
@@ -711,10 +716,10 @@
         self.sponsorBlockButton.frame = CGRectMake(0, 0, 40, 40);
         
         if([%c(YTPageStyleController) pageStyle]) { //dark mode
-            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblocksettings-20@2x.png"] forState:UIControlStateNormal];
+            [self.sponsorBlockButton setImage:[UIImage imageWithContentsOfFile:SBSettingsPath] forState:UIControlStateNormal];
         }
         else { //light mode
-            UIImage *image = [UIImage imageWithContentsOfFile:@"/var/mobile/Library/Application Support/iSponsorBlock/sponsorblocksettings-20@2x.png"];
+            UIImage *image = [UIImage imageWithContentsOfFile:SBSettingsPath];
             image = [image imageWithTintColor:UIColor.blackColor renderingMode:UIImageRenderingModeAlwaysTemplate];
             [self.sponsorBlockButton setImage:image forState:UIControlStateNormal];
             [self.sponsorBlockButton setTintColor:UIColor.blackColor];
@@ -839,3 +844,23 @@
         }
     }
 }
+
+
+//Bundle
+%ctor {
+NSString *resourcesBundlePath = [[NSBundle mainBundle] pathForResource:@"iSponsorBlock" ofType:@"bundle"];
+    if (resourcesBundlePath) {
+        NSBundle *resourcesBundle = [NSBundle bundleWithPath:resourcesBundlePath];
+        PlayerInfoIconSponsorBlockerPath = [resourcesBundle pathForResource:@"PlayerInfoIconSponsorBlocker256px-20@2x" ofType:@"png"];
+        SBStartPath = [resourcesBundle pathForResource:@"sponsorblockstart-20@2x" ofType:@"png"];
+        SBEndPath = [resourcesBundle pathForResource:@"sponsorblockend-20@2x" ofType:@"png"];
+        SBSettingsPath = [resourcesBundle pathForResource:@"sponsorblocksettings-20@2x" ofType:@"png"];
+    } 
+    else {
+        PlayerInfoIconSponsorBlockerPath = @"/Library/Application Support/iSponsorBlock.bundle/PlayerInfoIconSponsorBlocker256px-20@2x.png";    
+        SBStartPath = @"/Library/Application Support/iSponsorBlock.bundle/sponsorblockstart-20@2x.png"; 
+        SBEndPath = @"/Library/Application Support/iSponsorBlock.bundle/sponsorblockend-20@2x.png"; 
+        SBSettingsPath = @"/Library/Application Support/iSponsorBlock.bundle/sponsorblocksettings-20@2x.png"; 
+    }
+    %init;
+}
